test/e2e/antreapolicy_test.go

```diff
@@ -4438,7 +4509,25 @@ func (data *TestData) waitForANPRealized(t *testing.T, namespace string, name st
 	return nil
 }
 
-// waitForACNPRealized waits untils an ACNP is realized and returns, or times out. A policy is
+// waitForANPConditions waits until an ANP is processed but marked as pending with detailed conditions
+// and returns, or times out. A policy is considered processed when its Status has been updated so that
+// the ObservedGeneration matches the resource's Generation, the Phase is set to Pending, the Conditions
+// of Status is not empty.
+func (data *TestData) waitForANPConditions(t *testing.T, namespace string, name string, timeout time.Duration) error {
```

Comment:
The method name is generic but the comment and implementation is too specific to a scenario.

And why it needs to execute this? Doesn't `checkACNPStatus` already poll the status periodically until it matches or times out?

Reply:
Updated.